Wikipedia/Code/WebViewController.m

```diff
@@ -147,6 +150,34 @@ - (void)viewWillAppear:(BOOL)animated {
                                                object:nil];
     // should happen in will appear to prevent bar from being incorrect during transitions
     [self updateZeroState];
+
+    @weakify(self);
+    void (^ saveOpenArticleTitleBlock)(NSNotification*) = ^void (NSNotification* notification) {
```

Comment:
Or maybe I don't understand why did you change the logic to happen in viewdidappear instead of viewwilldissapear?

Basically I was thinking this would just be an extra piece of information on the history item. You save the scroll position and the current section title. If you want to use the fragment of the mwktitle that seems like a good idea, but I'm not sure why the timing of the logic needs to change. 

Basically why can't we just do your new logic in viewwilldissapear?